modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-channel-impl/src/main/java/com/liferay/headless/commerce/admin/channel/internal/resource/v1_0/CategoryDisplayPageResourceImpl.java

```diff
@@ -0,0 +1,229 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.headless.commerce.admin.channel.internal.resource.v1_0;
+
+import com.liferay.asset.kernel.model.AssetCategory;
+import com.liferay.commerce.product.constants.CPDisplayLayoutConstants;
+import com.liferay.commerce.product.exception.NoSuchCPDisplayLayoutException;
+import com.liferay.commerce.product.model.CPDisplayLayout;
+import com.liferay.commerce.product.model.CommerceChannel;
+import com.liferay.commerce.product.service.CPDisplayLayoutService;
+import com.liferay.commerce.product.service.CommerceChannelLocalService;
+import com.liferay.headless.commerce.admin.channel.dto.v1_0.CategoryDisplayPage;
+import com.liferay.headless.commerce.admin.channel.resource.v1_0.CategoryDisplayPageResource;
+import com.liferay.portal.kernel.search.BaseModelSearchResult;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.vulcan.dto.converter.DTOConverter;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
+import com.liferay.portal.vulcan.dto.converter.DefaultDTOConverterContext;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+/**
+ * @author Danny Situ
+ */
+@Component(
+	properties = "OSGI-INF/liferay/rest/v1_0/category-display-page.properties",
+	scope = ServiceScope.PROTOTYPE, service = CategoryDisplayPageResource.class
+)
+public class CategoryDisplayPageResourceImpl
+	extends BaseCategoryDisplayPageResourceImpl {
+
+	public void deleteCategoryDisplayPage(Long id) throws Exception {
+		_cpDisplayLayoutService.deleteCPDisplayLayout(id);
+	}
+
+	public CategoryDisplayPage getCategoryDisplayPage(Long id)
+		throws Exception {
+
+		CPDisplayLayout cpDisplayLayout =
+			_cpDisplayLayoutService.fetchCPDisplayLayout(id);
+
+		if (cpDisplayLayout == null) {
+			throw new NoSuchCPDisplayLayoutException();
+		}
```

Comment:
I think you can pass the id directly to the DTO Converter that uses the get method of the service, this way if the resource doesn't exist, it throws the nofoundexception that will be managed by the exception mapper and the code here will be more readable